analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ArrayPassedAsParams.vb

```diff
@@ -0,0 +1,66 @@
+﻿
+Public Class Program
+    Public Sub Base(ByVal myArray As String())
+        Method(New String() {"s1", "s2"}) ' Noncompliant {{Remove this array creation and simply pass the elements.}}
+        '      ^^^^^^^^^^^^^^^^^^^^^^^^^
+        Method(New String() {"s1"}) ' Noncompliant
+        Method("s1")               ' Compliant
+        Method("s1", "s2")         ' Compliant
+        Method(myArray)            ' Compliant
+        Method(New String(11) {})  ' Compliant
```

Comment:
VB is funny
* Valid: `New String() { }` 
* Valid: `New String(3) { }` 
* Invalid: `New String(3) { "s" }` *BC30567: Array initializer is missing 3 elements.*
* Valid: `New String(3) { "s", "s", "s", "s" }`
* Valid: `{ "s", "s", "s", "s" }`

To be really precise, we need to check for more cases:
* Emtpy CollectionInitializer `{ }` -> NonCompliant (Missing test case)
* CollectionInitializer with values `{ "s", "s", "s", "s" }` -> NonCompliant (Missing test case)
* Array bounds specified and Collection initializer empty `new String(2) { }` -> Compliant (Covered)
* Array bounds specified and Collection initializer not empty `new String(2) { "s", "s", "s" }` -> Noncompliant (Not covered)
* Array bounds not specified and Collection initializer empty `new String() { }` -> Noncompliant (Not Covered)
* Array bounds not specified and Collection initializer not empty `new String() { "s" }` -> Noncompliant (Covered)

I often use parameterized tests for such scenarios because our the test files get pretty big if you want to cover each and every combination in there. The test files are also a nightmare if you want to debug a specific case only.

This could look something like this:

In analyzers/tests/SonarAnalyzer.UnitTest/Rules/ArrayPassedAsParamsTest.cs
```cs
        [DataTestMethod]
        [DataRow("{ }", false)]
        [DataRow("{ "s", "s", "s", "s" }", false)]
        [DataRow("new String(2) { }", true)]
        // ...
        public void VBCollectionInitializerSyntaxTests(string arrayCreation, bool compliant)
        {
            var code = $$"""
            Public Class C
                Public Sub M()
                    ParamsMethod({{arrayCreation}}) ' {{compliant ? "Compliant": "Noncompliant"}}
                End Sub

                Public Sub ParamsMethod(ParamArray p As String())
                End Sub
            End Class
            """;
            var builder = builderVB.AddSnippet(code);
            if (compliant)
            {
                builder.VerifyNoIssueReported();
            }
            else
            {
                builder.Verify();
            }
        }
```

Reply:
`New String(3) { "s", "s", "s", "s" }` 😕
This is brilliant 💯

Anyway, I covered all of these, great job identifying all those cases!
I like the parametrized snippet test, pretty useful for debugging as you said.